### PR TITLE
Add expression nullable support

### PR DESCRIPTION
**Pull request by junli1026**
I hereby agree to the terms of the CLA available at: https://databend.rs/policies/cla/

## Summary

Add expression nullable support.

## Changelog

- Improvement

## Related Issues

Fixes #issue

## Test Plan

Stateless Tests

### CODE REVIEW

File: tests/suites/0_stateless/05_0000_ddl_create_tables.sql [hunk 1]
```diff
@@ -19,15 +19,24 @@ DROP TABLE IF EXISTS t2;
...
+
+SELECT '====BEGIN TEST CREATE TABLE AS SELECT STATEMENT====';
 -- test 'create table as select' statement, expect db2.test3 has the data from db1.test1 with casting
 CREATE TABLE db2.test3(x Varchar, y Varchar) ENGINE=fuse AS SELECT * FROM db1.test1;
 SELECT x FROM db2.test3;
+SELECT '====END TEST CREATE TABLE AS SELECT STATEMENT====';
+
 -- clean up test databases
 DROP DATABASE db1;
 DROP DATABASE db2;
```
Comment (sundy-li) [inline tests/suites/0_stateless/05_0000_ddl_create_tables.sql:42]: new line needed.

**Comment by sundy-li** [comment]
Nullable is annoyment.

ClickHouse takes Nullable as a separated datatype.
Yet MySQL takes Nullable as a property for datatype.

So in the function, there are two kinds of ways to handle nullable:

1. if args are not nullable, the result should not be nullable except some special functions (isNull.. )
2. if args are not nullable, the result can be nullable.

I prefer the first one, thus we can control the return type.

**Comment by sundy-li** [comment]
`query group by nullable(col)` may take much more memory/costs than `query group by col`.

**Comment by junli1026** [comment]
> Nullable is annoyment.
> 
> ClickHouse takes Nullable as a separated datatype. Yet MySQL takes Nullable as a property for datatype.
> 
> So in the function, there are two kinds of ways to handle nullable:
> 
> 1. if args are not nullable, the result should not be nullable except some special functions (isNull.. )
Sure, will address. Do we have an example of the 'isNull' ? --  the inputs are all not nullable but output is null ? 

> 2. if args are not nullable, the result can be nullable.
> 
> I prefer the first one, thus we can control the return type.

**Comment by sundy-li** [comment]
MySQL can produce nullable result by none nullable args.

 
image

**Comment by junli1026** [comment]
> MySQL can produce nullable result by none nullable args.
> 
> image

Right, that is why I just check the function's nullable() return. My initial thoughts are just like, checking the input, if any of them is nullable, the output should be nullable.

**Review by sundy-li — APPROVED** [review]

**Comment by BohuTANG** [comment]
Thank you @junli1026 !

**Pull request merged**